maps/src/main/java/com/gluonhq/impl/maps/tile/osm/CachedOsmTileRetriever.java

```diff
@@ -46,16 +46,16 @@
 import java.util.logging.Level;
 import java.util.logging.Logger;
 
-public class CachedOsmTileRetriever extends OsmTileRetriever {
+public abstract class CachedOsmTileRetriever extends OsmTileRetriever {
 
     private static final Logger logger = Logger.getLogger(CachedOsmTileRetriever.class.getName());
-    private static final int TIMEOUT = 5000;
+    private static  final int TIMEOUT = 5000;
 
-    static File cacheRoot;
-    static boolean hasFileCache;
-    static CacheThread cacheThread = null;
+     File cacheRoot;
+     boolean hasFileCache;
+     CacheThread cacheThread = null;
 
-    static {
+     {
```

Comment:
Not a huge fan of double brace initialization. May be add a constructor instead?

Reply:
I am not sure if the logic was appropriate before you made the change., but now we have different instances of `CacheThread` writing to the same cache directory.

Reply:
You mean there were 2 instances of CacheThread because the sample created 2  CachedOsmTileRetriever, right?
If this was the issue, I just added a name to the CachedOsmTileRetriever constructor and it is used to build the cache path.

Reply:
I am not a big fan of the implementation in `CachedOsmTileRetriever`. If you check `CacheThread`, you will find that it has an indefinite while loop and it keep on running. The `deactivate()` which is supposed to break the while loop is never called. The whole class needs to be re-written IMO.

Reply:
Can you please remove the change where you added a string name as suffix for cache directory? It is better to keep the class as it is. I will log an issue with the shortcomings on the class.

Reply:
I have reverted the commit that added suffix to cache